src/pudl/transform/eia861.py

```diff
@@ -776,22 +776,21 @@ def balancing_authority_assn(tfr_dfs):
         transform functions.
 
     """
-    # The dataframes from which to compile BA-Util-State associations
-    other_dfs = [
-        tfr_dfs["sales_eia861"],
-        tfr_dfs["demand_response_eia861"],
-        tfr_dfs["advanced_metering_infrastructure_eia861"],
-        tfr_dfs["dynamic_pricing_eia861"],
-        tfr_dfs["net_metering_eia861"],
+    # These are structural rather than "data" tables, and should not be searched
+    structural_dfs = [
+        "balancing_authority_eia861",
+        "service_territory_eia861",
```

Comment:
The `balancing_authority_eia861` table isn't well structured at this point -- it's still a mix of two different structures from the early and late years, and once it has its final structure, it doesn't have any of these relationships in it.

The `service_territory_eia861` table connects dates, utilities, and states, but in an expansive way -- it's *all* of the possible combinations.  And the point here is to be able to limit what subset of that table is selected based on which combinations of state/util/date have *actually* been observed in the data tables. So I'm purposefully excluding the structural tables here so that the relationships in them *do not* get extracted.

Reply:
Yes, I see that these two are not getting extracted. And this makes sense. They sound like relationship/assn tables instead of structural? Maybe rename them to be assn_dfs or relational_dfs or just dfs_not_to_seach or something.

Reply:
Okay I'll call them `non_data_dfs`